src/client/_clientwindow.py

```diff
@@ -1060,6 +1043,9 @@ def _tabChanged(self, tab, curr, prev):
     def mainTabChanged(self, curr):
         self._tabChanged(self.mainTabs, curr, self._main_tab)
         self._main_tab = curr
+        # the tab change works after client has kinda initialized
+        if self.game_announcer.delay_friend_events:
+            self.game_announcer.delayed_friend_events()
 
```

Comment:
Sad, the client start announcements can still get lost, if you click Chat 'to early' while chat is still building up.
So either we find another 'chat is ready for lines'-signal place, or - the ugly way - add an extra delay in/before delayed_friend_events.

Reply:
Let's not complicate things just yet. The ideal way really would be to separate chat from widgets - as in make a "chat model" with representations of channels, chatters, lines etc, and have widgets only show changes. We'd then initialize and hook up model stuff earlier and not worry about anything being lost. I did a bit of work towards this, but stalled a bit, see https://github.com/Wesmania/client/tree/chat-rewrite.

Reply:
If it's not working reliably, that's not complicating things, that's just sad, and should be fixed if possible.
We don't know what percentage of users click on the Chat-Tab right away after client start.

Reply:
It's not obvious to me which point in the code can be considered "chat widget fully loaded". You're welcome to hunt one down.

Reply:
well, I tried that for some time already, and didn't find any better place ... so far.
I might try some more ...

Reply:
I added a 2nd holdof in _chadwidget  - on_join, now the announcement is delivered, but in grey (insteat of friend color), so something is still not ready. (the announced player has a name, but is not yet in chatter, still at the 'one man show' level)

Reply:
well, found a solution over addChatter in channel.py.
There might be room for improvement in elegance, suggestions are welcome.